File: src/main/webapp/js/services/contractService.js
var services = angular.module('contractService',['ngResource']);

services.factory('contractService', function($resource) {
	return $resource('rest/Contract/:action',{},
			{
		getStudent : {
			method : 'POST', isArray: true,
			params: {'action':'getEmpByName'}
		},
		getDeptList :{
			method: 'GET',isArray:true,
			params:{'action':'getDeptList'}
		},
		searchByJobTitle : {
			method: 'POST', isArray:true,
			params: {'action':'searchByJobTitle'}
		},
		editContractStudent : {
			method : 'POST',isArray : false,
			params : {'action':'editContractStudent'}
		},
		getStudentInImage : {
			method : 'POST',isArray : true,
			params : {'action':'getStudentInImage'}
		}
	});
});

File: src/main/java/com/project/service/ContractService.java
package com.project.service;

import java.util.List;

import com.project.model.Contract;

public interface ContractService {

	List<Contract> getEmpByName(Contract contract);

	List<String> getDeptList();

	List<Contract> searchByJobTitle(String department);

	int editContractStudent(Contract contract);

	List<Contract> getStudentInImage(Contract contract);

	List<String> getjobTitle();

	List<Contract> searchByNameString(Contract contract);

	int deletecontract(Contract contract);

	

}
